### PR TITLE
VueUse functions search extension added

### PR DESCRIPTION
**Pull request by noopurphalak**
## Description

I have added a new raycast extension which makes it easy for the user to search vueuse functions and navigate to [vueuse](https://vueuse.org/functions.html) website for the documentation on the selected function.

## Screencast

Screenshot 2022-02-27 at 10 11 27 PM

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)

### CODE REVIEW

**Comment by noopurphalak** [comment]
@pernielsentikaer Can you please review this PR?

File: extensions/vueuse-functions/package.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  "name": "vueuse-functions",
+  "title": "VueUse Functions",
+  "description": "Search VueUse Functions",
+  "icon": "vue-use.png",
+  "author": "noopurphalak",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "index",
+      "title": "Search VueUse Functions",
```
Comment (pernielsentikaer) [inline extensions/vueuse-functions/package.json:12]:
```suggestion
      "title": "Search Documentation",
      "subtitle": "VueUse",
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @noopurphalak 👋 

Sorry for the longer response time 💯 It's looking good 🔥 

I left some feedback and suggestions for you to address:

- Did you consider adding multiple versions, so it's possible to search older versions? (Could be new commands)
- You could show all items in the list when entering the extension, instead of just “No Results”

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by noopurphalak** [comment]
> Hi @noopurphalak 👋
> 
> Sorry for the longer response time 💯 It's looking good 🔥
> 
> I left some feedback and suggestions for you to address:
> 
> * Did you consider adding multiple versions, so it's possible to search older versions? (Could be new commands)
> * You could show all items in the list when entering the extension, instead of just “No Results”
> 
> Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

Hi @pernielsentikaer, thanks for reviewing my PR. Do you have any suggestions on how I can implement search for various versions? Currently, the way this extension is implemented, I am relying on the JSON data I scrapped from the website. For multiple versions, I will need to maintain multiple JSON files as there is no documentation endpoint that they provide.

Will also initially load all the functions for smooth user experience as you suggested.

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @noopurphalak 👋 

In the newest version, `1.30.2`, of `@raycast/api` we introduced [search bar dropdown](https://developers.raycast.com/changelog#1.30.0-2022-03-09) - with this you should be able to do some magic with a fancy dropdown where it's possible to change version, it will look something like this:

vu1V2XdfSXXAx4Pg

You can read more about [search bar dropdown here](https://developers.raycast.com/api-reference/user-interface/list#dropdown)

The way you're doing it is also how the most other documentations are handling it at the moment when it's not possible to use APIs

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by sxn** [comment]
hi @noopurphalak - I've updated the PR to address a couple of @pernielsentikaer 's suggestions. Additionally I've removed the restriction that the user _must_ type something before any functions are shown.

Regarding how to get multiple versions: it depends on you, really:
- you could crawl the website at runtime and build the list on the fly. The advantage would be that the list would always be up to date. Disadvantages: slower, lots of unnecessary requests (not sure how often the website is expected to change), prone to breaking if the website ever changes its structure
- you could go the same route as the [.new extension](https://github.com/raycast/extensions/pull/1254#pullrequestreview-928038157): crawl the website & create the corresponding `.json` files at compile time and bundle them with the app. That way the extension should be fast, it shouldn't break even if the website changes its structure. You run the risk that maybe a new hook is not part of the extension, but depending on how often the website is expected to change, that might not actually be a problem in practice.

As implementing version-select is a bigger task, I think it's something that can be done in a subsequent PR. What do you think?

**Comment by pernielsentikaer** [comment]
Hi @noopurphalak and @sxn 👋 

Let's get the first version out, and then we can look into versioning later 🚀

**Pull request merged**